addons/mail/static/src/core/common/store_service.js

```diff
@@ -183,6 +183,34 @@ export class Store extends BaseStore {
         return tab === "chat" ? ["chat", "group"] : [tab];
     }
 
+    async handleClickOnLink(ev, thread) {
+        const model = ev.target.dataset.oeModel;
+        const id = Number(ev.target.dataset.oeId);
+        if (ev.target.closest(".o_channel_redirect") && model && id) {
+            ev.preventDefault();
+            const thread = this.Thread.insert({ model, id });
+            this.env.services["mail.thread"].open(thread);
+            return true;
+        } else if (ev.target.closest(".o_mail_redirect")) {
```

Comment:
You might want to check `&& id` here as well if you're going that way with adding missing checks.